src/main/java/com/scalar/db/benchmarks/ycsb/LoadRunner.java

```diff
@@ -0,0 +1,130 @@
+package com.scalar.db.benchmarks.ycsb;
+
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.NAMESPACE_PRIMARY;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.NAMESPACE_SECONDARY;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.getLoadBatchSize;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.getLoadConcurrency;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.getLoadOverwrite;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.getPayloadSize;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.getRecordCount;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.prepareGet;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.preparePut;
+import static com.scalar.db.benchmarks.ycsb.YcsbCommon.randomFastChars;
+
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.DistributedTransactionManager;
+import com.scalar.db.api.Get;
+import com.scalar.db.api.Put;
+import com.scalar.db.benchmarks.Common;
+import com.scalar.db.exception.transaction.AbortException;
+import com.scalar.db.exception.transaction.TransactionException;
+import com.scalar.kelpie.config.Config;
+import io.github.resilience4j.retry.Retry;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.IntStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoadRunner {
+  private static final Logger LOGGER = LoggerFactory.getLogger(LoadRunner.class);
+  private final DistributedTransactionManager manager;
+  private final int id;
+  private final int concurrency;
+  private final int recordCount;
+  private final char[] payload;
+  private final int batchSize;
+  private final boolean overwrite;
+
+  public LoadRunner(Config config, int threadId) {
+    this.id = threadId;
+    manager = Common.getTransactionManager(config);
+    concurrency = getLoadConcurrency(config);
+    batchSize = getLoadBatchSize(config);
+    recordCount = getRecordCount(config);
+    payload = new char[getPayloadSize(config)];
+    overwrite = getLoadOverwrite(config);
+  }
+
+  public void run() {
+    run(false);
+  }
+
+  public void runForMultiStorage() {
+    run(true);
+  }
+
+  private void run(boolean forMultiStorage) {
+    int numPerThread = (recordCount + concurrency - 1) / concurrency;
```

Comment:
Honestly, I didn't care about this part very much, but I think this is needed to avoid losing the remainder. When we have 10 records and 3 threads, the assignment will be three each due to the round down.

Reply:
Ah, thanks! You're right